tests/integration/ha/helpers.py

```diff
@@ -504,24 +514,62 @@ async def start_and_check_continuous_writes(ops_test: OpsTest, unit_ip: str, app
     )
     writer = ContinuousWrites(ops_test, app, initial_count=initial_count)
     await writer.start()
-    time.sleep(10)
-    result = await writer.stop()
-    return result.count > initial_count
+    time.sleep(60)
+    # Ensure we have writes happening and the index is consistent at the end
+    await assert_continuous_writes_increasing(writer)
+    await assert_continuous_writes_consistency(ops_test, writer, app)
+    # Clear the writer manually, as we are not using the conftest c_writes_runner to do so
+    await writer.clear()
 
 
-async def backup_cluster(ops_test: OpsTest, leader_id: int) -> int:
+async def create_backup(ops_test: OpsTest, leader_id: int, unit_ip: str) -> int:
     """Runs the backup of the cluster."""
     action = await run_action(ops_test, leader_id, "create-backup")
-    assert action.status == "completed"
     logger.debug(f"create-backup output: {action}")
 
-    await wait_for_backup(ops_test, leader_id)
+    await wait_for_backup_system_to_settle(ops_test, leader_id, unit_ip)
+    assert action.status == "completed"
+    assert action.response["status"] == "Backup is running."
     return int(action.response["backup-id"])
 
 
-async def restore_cluster(ops_test: OpsTest, backup_id: int, unit_ip: str, leader_id: int) -> bool:
-    action = await run_action(ops_test, leader_id, "restore", params={"backup-id": backup_id})
+async def restore(ops_test: OpsTest, backup_id: int, unit_ip: str, leader_id: int) -> bool:
+    """Restores a backup."""
+    id = backup_id
+    if not isinstance(backup_id, int):
+        id = int(backup_id)
+    action = await run_action(ops_test, leader_id, "restore", params={"backup-id": id})
     logger.debug(f"restore output: {action}")
```

Comment:
also it is necessary to check that the writes are actually restored

Reply:
These are helpers, they do not run `assert`.